data/styles/clear/include/Basemap_label.mapcss

```diff
@@ -406,7 +406,9 @@ area|z16-[leisure=garden][garden:type!=residential][name]::int_name
 */
 area|z14-[leisure=park][!name],
 area|z14-[landuse=forest][!name],
-area|z16-[leisure=garden][!name],
+area|z16-[leisure=garden][garden:type!=residential][!name],
+{text: none}
+area|z16-[leisure=garden][garden:type=residential],
 {text: none}
```

Comment:
Why so fancy selector? Enough here, no?
```
area|z16-[leisure=garden][garden:type=residential],
{text: none}
```

Reply:
nope, doesn't work like this...
we need to reset the caption for `leisure-garden`[!name] also

(and overall the "dynamic tags" [name]/[!name]/et. inheritance is crooked and buggy)

Reply:
I see no sense here. Remove text style for Features without name :)

AFAIR, we added this [name] selector to draw or not draw symbols. But why captions? Just declare text in base style and it won't be visible anyway for Features without names :)

Reply:
1. Your suggestion leads to removal of z16 & z17 area fills from `leisure-garden` ("dynamic tags" processing is crooked!)
2. Why store and process in runtime the drules that will never be used? [name]/[!name] is used for this case already anyway, why not optimize the captions also?

Reply:
Well, ok if it is crooked. ATM [!name] && {text: none} looks strange :)